server/raft.go

```diff
@@ -3434,16 +3436,6 @@ func (n *raft) processAppendEntry(ae *appendEntry, sub *subscription) {
 				return
 			}
 
-			// Check if only our terms do not match here.
-			if ae.pindex == n.pindex {
```

Comment:
This condition could never be hit before, due to the condition above:
```go
// Check if this is a lower or equal index than what we were expecting.
if ae.pindex <= n.pindex {
```

Now moved up so the `pterm` can be corrected, otherwise we could spin.

Already covered by the test `TestJetStreamClusterStreamCatchupNoState`. Previously it used a different way to pass, likely truncating a bit too much and eventually converging. Now it only corrects the `pterm` as truncating is not needed.